src/topology/perfect.lean

```diff
@@ -267,24 +268,21 @@ open cantor_scheme
 
 /-- Any nonempty perfect set in a complete metric space admits a continuous injection
 from the cantor space, `ℕ → bool`. -/
-theorem perfect.exists_nat_bool_injection [complete_space α]
-  (hC : perfect C) (hnonempty : C.nonempty) :
+theorem perfect.exists_nat_bool_injection [complete_space α] :
   ∃ f : (ℕ → bool) → α, (range f) ⊆ C ∧ continuous f ∧ injective f :=
 begin
-  let u : ℕ → ennreal := λ n, n⁻¹,
-  have upos : ∀ n, 0 < (u n) := λ n, by simp,
+  obtain ⟨u, -, upos', hu⟩ := exists_seq_strict_anti_tendsto' (zero_lt_one' ℝ≥0∞),
+  have upos := λ n, (upos' n).1,
   let P := subtype (λ E : set α, perfect E ∧ E.nonempty),
-  choose C0 C1 h0 h1 hdisj using @perfect.small_diam_splitting α infer_instance,
-  change ∀ {C} {hC : perfect C} {hnonempty : C.nonempty} {ε : ennreal} {ε_pos : 0 < ε}, _ at h0,
-  change ∀ {C} {hC : perfect C} {hnonempty : C.nonempty} {ε : ennreal} {ε_pos : 0 < ε}, _ at h1,
-  change ∀ {C} {hC : perfect C} {hnonempty : C.nonempty} {ε : ennreal} {ε_pos : 0 < ε}, _ at hdisj,
+  choose C0 C1 h0 h1 hdisj using λ {C : set α} {hC : perfect C} {hnonempty : C.nonempty}
+    {ε : ℝ≥0∞} {hε : 0 < ε}, hC.small_diam_splitting hnonempty hε,
   let DP : list bool → P := λ l,
   begin
     induction l with a l ih, { exact ⟨C, ⟨hC, hnonempty⟩⟩ },
     cases a,
-    { use C0 ih.property.1 ih.property.2 (u l.length.succ) (upos _),
+    { use @C0 _ ih.property.1 ih.property.2 _ (upos l.length.succ),
```

Comment:
The point of me giving you more explicit arguments was precisely to avoid you having to use `@` here.

Reply:
This is why the `C0` and `C1` originally used explicit arguments. But the point of the original ugly block of `change`'s was to make the arguments for `h0`, `h1`, and `hdisj` implicit, which I think is definitely desirable.

Reply:
Interesting. Could you explain?

Reply:
The idea of hygiene is to reduce the likelihood your proof will break in the future. With what you had before, changing the arguments to `perfect.small_diam_splitting` would have resulted in the breakage of at least four lines, and the fix would have been to shuffle binders in the lengthy `change`. Now, if someone changes `perfect.small_diam_splitting`, the fix will be to update a single line, the `choose`.